pkg/controller/pravega/pravega_segmentstore.go

```diff
@@ -174,13 +167,39 @@ func makeSegmentstorePodSpec(p *api.PravegaCluster) corev1.PodSpec {
 	return podSpec
 }
 
+func MakeSegmentStoreVolumes(p *api.PravegaCluster) []corev1.Volume {
+	volumes := []corev1.Volume{
+		{
+			Name: heapDumpName,
+			VolumeSource: corev1.VolumeSource{
+				EmptyDir: &corev1.EmptyDirVolumeSource{},
+			},
+		},
+	}
+	if p.Spec.Pravega.Tier2.Ecs != nil && strings.TrimSpace(p.Spec.Pravega.Tier2.Ecs.Certificates) != "" {
+		volumes = append(volumes, corev1.Volume{
+			Name: "ecs-certs",
+			VolumeSource: corev1.VolumeSource{
+				Secret: &corev1.SecretVolumeSource{SecretName: p.Spec.Pravega.Tier2.Ecs.Certificates},
+			},
+		})
+	}
+	return volumes
+}
+
 func MakeSegmentStoreVolumeMount(p *api.PravegaCluster) []corev1.VolumeMount {
 	volumeMount := []corev1.VolumeMount{
 		{
 			Name:      heapDumpName,
 			MountPath: heapDumpDir,
 		},
 	}
+	if p.Spec.Pravega.Tier2.Ecs != nil && strings.TrimSpace(p.Spec.Pravega.Tier2.Ecs.Certificates) != "" {
+		volumeMount = append(volumeMount, corev1.VolumeMount{
+			Name:      "ecs-certs",
+			MountPath: "/etc/ssl/certs/java/ecs-certs/",
```

Comment:
It would be good to define constants for hard-coded values.

Reply:
Yes, fixed

Reply:
You perhaps missed pushing your changes?